dep0/lib/03_typecheck/src/beta_reduction.cpp

```diff
@@ -207,11 +217,35 @@ bool beta_normalize(body_t& body)
                     else
                         return body.stmts.erase(it);
                 }
+                else if (is_mutable(if_.cond))
+                    // if the condition is mutable we don't want to mess around with things
```

Comment:
```suggestion
                    // if the condition is mutable we don't want to mess around with things;
                    // this could be improved in future but for now it's fine like this
```